models/models.go

```diff
@@ -120,6 +122,15 @@ type StepGroupModel struct {
 	Versions            map[string]StepModel `json:"versions,omitempty" yaml:"versions,omitempty"`
 }
 
+// Key: platform, as in runtime.GOOS + runtime.GOARCH
+// Examples: darwin-arm64, linux-amd64
+type Executables map[string]Executable
+
+type Executable struct {
+	Url string `json:"url,omitempty" yaml:"url,omitempty"`
```

Comment:
I am not convinced it is a good thing to add the URL in the step.ym, as we could keep it less transparent. For example we can host binaries in different storage providers, caches.

An other complication would be if we decide to upgrade go version and recompile the stepsm then the hash could change. Not sure if we want to do this tough.

Reply:
Forgot to mention that this is only present in the steplib-submitted `step.yml`, just like the `source` block (which is not a great design, but it is what it is). So I imagine this `executable` block to be filled by the steplib submission tooling, just like it happens today with the `source` block. This way, a given URL is only associated with a tagged step release, not the step repo itself.

Reply:
> An other complication would be if we decide to upgrade go version and recompile the stepman then the hash could change. Not sure if we want to do this tough.

I don't get this part though. Upgrade the go version where exactly?

Reply:
I imagine the Go compiler upgrade like this:
- step is at v1.0.0 -> it has a tagged release URL and a matching hash
- Go upgrade, and any other tooling changes in the step repo
- Cut a new step release at v1.0.1 -> new tagged release URL and a new hash